pkg/tfgen/generate_go.go

```diff
@@ -438,6 +438,12 @@ func (g *goGenerator) emitResourceType(mod *module, res *resourceType) error {
 	if res.doc != "" {
 		g.emitDocComment(w, res.doc, res.docURL, "")
 	}
+	if !res.IsProvider(){
+		deprecationMessage := g.getDeprecationDetails(res.info)
+		if deprecationMessage != "" {
+			w.Writefmtln("%s", deprecationMessage)
```

Comment:
my thinking on deprecation is that it should override normal docs.  so we should just check this first and skip normal docs if we have it.